x-pack/plugins/maps/public/classes/sources/es_geo_grid_source/es_geo_grid_source.js

```diff
@@ -112,7 +139,11 @@ export class ESGeoGridSource extends AbstractESAggSource {
   }
 
   getGeoGridPrecision(zoom) {
-    const targetGeotileLevel = Math.ceil(zoom) + this._getGeoGridPrecisionResolutionDelta();
+    const delta = this._getGeoGridPrecisionResolutionDelta();
```

Comment:
What do you think about removing `this._descriptor.resolution === GRID_RESOLUTION.SUPER_FINE` case from `_getGeoGridPrecisionResolutionDelta` and just doing that check directly in `getGeoGridPrecision`. So _getGeoGridPrecisionResolutionDelta is unchanged and `getGeoGridPrecision` becomes

```
getGeoGridPrecision(zoom) {
  if (this._descriptor.resolution === GRID_RESOLUTION.SUPER_FINE) {
    // The target-precision needs to be determined server side.
    return NaN;
  }
  const targetGeotileLevel = Math.ceil(zoom) + this._getGeoGridPrecisionResolutionDelta();
  return Math.min(targetGeotileLevel, MAX_GEOTILE_LEVEL);
}
```